e2etests/test/empty_messages.js

```diff
@@ -0,0 +1,210 @@
+                done(err);
+              } else {
+                assert.equal(result.constructor.name, 'MessageCompleted');
+                if (foundTheMessage) {
+                  deviceClient.removeAllListeners('message');
+                  testRendezvous.imDone(deviceClientParticipant);
+                }
+              }
+            });
+          });
+          debug('about to open the service client');
+          serviceClient.open(function (serviceErr) {
+            debug('At service client open callback - error is:' + serviceErr);
+            if (serviceErr) {
+              done(serviceErr);
+            } else {
+              testRendezvous.imIn(serviceClientParticipant);
+              serviceClient.send(provisionedDevice.deviceId, message, function (sendErr) {
+                debug('At service client send callback - error is: ' + sendErr);
+                if (sendErr) {
+                  done(sendErr);
+                } else {
+                  testRendezvous.imDone(serviceClientParticipant);
+                }
+              });
+            }
+          });
+        }
+      });
+    });
+  });
+
+  describe('Over ' + deviceTransport.name + ' using device/eventhub clients - messaging', function () {
+    this.timeout(18000);
+
+    var deviceClient, ehClient, ehReceivers, provisionedDevice;
+
+    before(function (beforeCallback) {
+      createDeviceMethod(function (err, testDeviceInfo) {
+        provisionedDevice = testDeviceInfo;
+        beforeCallback(err);
+      });
+    });
+
+    after(function (afterCallback) {
+      DeviceIdentityHelper.deleteDevice(provisionedDevice.deviceId, afterCallback);
+    });
+
+    beforeEach(function () {
+      ehClient = eventHubClient.fromConnectionString(hubConnectionString);
+      deviceClient = createDeviceClient(deviceTransport, provisionedDevice);
+      ehReceivers = [];
+    });
+
+    afterEach(function (done) {
+      closeDeviceEventHubClients(deviceClient, ehClient, ehReceivers, done);
+    });
+    it('Device sends a message of zero size and it is received by the service', function (done) {
+      var deviceClientParticipant = 'deviceClient';
+      var ehClientParticipant = 'ehClient';
+      var testRendezvous = new Rendezvous(done);
+      this.timeout(120000);
+      //
+      // The size of non payload is calculated based on the value of the messageId value size
+      // and then on the application property value size plus one for the application property name.
+      //
+      var uuidData = uuid.v4();
+      var message = new Message('');
+      message.messageId = uuidData;
+      ehClient.open()
+        .then(function() {testRendezvous.imIn(ehClientParticipant);})
+        .then(ehClient.getPartitionIds.bind(ehClient))
+        .then(function (partitionIds) {
+          return partitionIds.map(function (partitionId) {
+            return ehClient.createReceiver('$Default', partitionId,{ 'startAfterTime' : Date.now()}).then(function(receiver) {
```

Comment:
>d,{  [](start = 65, length = 4)

nit, spacing

Reply:
Fixed.

---
In reply to: [227565914](https://github.com/Azure/azure-iot-sdk-node/pull/401#discussion_r227565914) [](ancestors = 227565914)